### PR TITLE
CN-5514 margin call flow

### PR DESCRIPTION
**Pull request by ShamanShaulic**
## Checklist (Mandatory):

- [x] I have named the PR correctly (Check PR1)
- [x] Check the commit’s or even all commits message styles matches our requested structure. (issue number, etc.) (Check CR10)
- [x] I have tested new code locally (Check CR3)
- [x] I have done a smoke test of the larger feature locally (Check CR3)
- [x] I have added at least two reviewers (Check PR5)
- [x] I have assigned the PR to myself (Check PR4)
- [x] I have merged develop (parent) branch into my feature branch and resolved all conflicts (Check PR7)

## Checklist (Optional):

- [ ] I have added necessary documentation, including instructions how to test (Check CR8)
- [ ] I have added tests that prove my fix/implementation is effective or that my feature works
- [ ] I have written the code according to the adopted style guide? (Check CR7)

## Additional notes

> What does this implement/fix? Explain your changes. (optional)
> Are there any scripts that need running (eg. yarn install)?
> Are there any edge cases that need checking?
> Are there any specific test instructions?

[MVPWorkshop Pull Request & Code Review Guidelines](https://docs.google.com/document/d/1TcuU1JBhlGZLtpy0osSNinXQ2piP_yrDuRcXdqaOS_U/edit#heading=h.5b6kpdr4btq8)

### CODE REVIEW

File: app/components/modals/LoanAlertsModals/LoanAlertsPayoutPrincipalModal/LoanAlertsPayoutPrincipalModal.js [hunk 1]
```diff
@@ -1,59 +1,153 @@
...
+        <View style={style.buttonsWrapper}>
+          <CelModalButton
+            buttonStyle={"green"}
+            position={"single"}
+            loading={isLoadingClose}
+            onPress={content.onPress}
+          >
+            {content.buttonText}
+          </CelModalButton>
+          {/* <CelModalButton*/}
```
Comment (fjovakaric) [inline app/components/modals/LoanAlertsModals/LoanAlertsPayoutPrincipalModal/LoanAlertsPayoutPrincipalModal.js:140]: remove comments

File: app/components/modals/LoanAlertsModals/LoanAlertsPayoutPrincipalModal/LoanAlertsPayoutPrincipalModal.js [hunk 2]
```diff
@@ -1,59 +1,153 @@
...
+          {/* >*/}
+          {/*  {content.buttonTextTwo}*/}
+          {/* </CelModalButton>*/}
+        </View>
+      </CelModal>
     );
   }
 }
-
-LoanAlertsPayoutPrincipalModal.propTypes = {
```
Comment (fjovakaric) [inline app/components/modals/LoanAlertsModals/LoanAlertsPayoutPrincipalModal/LoanAlertsPayoutPrincipalModal.js:53]: why is this removed?

File: app/components/modals/LoanApplicationSuccessModal/LoanApplicationSuccessModal.js [hunk 3]
```diff
@@ -31,8 +40,28 @@ class LoanApplicationSuccessModal extends Component {
     actions.closeModal();
   };
 
+  // setupAutomaticInterestPayment = async m => {
```
Comment (fjovakaric) [inline app/components/modals/LoanApplicationSuccessModal/LoanApplicationSuccessModal.js:43]: remove comments

File: app/components/modals/MarginCallConfirmModal/MarginCallConfirmModal.styles.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  buttonsWrapper: {
+    justifyContent: "flex-end",
+    marginTop: 20,
+    height: 50,
+  },
+};
+
+const themed = {
+  light: {
+    card: { color: "#F3F3F3" },
```
Comment (fjovakaric) [inline app/components/modals/MarginCallConfirmModal/MarginCallConfirmModal.styles.js:23]: use new colors

File: app/components/modals/MarginCallConfirmModal/MarginCallConfirmModal.styles.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  },
+
+  dark: {
+    card: { color: STYLES.COLORS.MEDIUM_GRAY },
```
Comment (fjovakaric) [inline app/components/modals/MarginCallConfirmModal/MarginCallConfirmModal.styles.js:27]: use new colors

File: app/components/molecules/CollateralCoinCard/CollateralCoinCard.js [hunk 6]
```diff
@@ -291,36 +61,60 @@ class CollateralCoinCard extends Component {
...
+                    flexDirection: "row",
+                    alignItems: "center",
+                    marginTop: 10,
+                  }}
+                >
+                  <Icon
+                    name={"CirclePlus"}
+                    width={20}
+                    height={20}
+                    fill={STYLES.COLORS.CELSIUS_BLUE}
```
Comment (fjovakaric) [inline app/components/molecules/CollateralCoinCard/CollateralCoinCard.js:104]: use new colors

File: app/components/molecules/CollateralCoinCard/CollateralCoinCard.js [hunk 7]
```diff
@@ -291,36 +61,60 @@ class CollateralCoinCard extends Component {
...
+                  />
+                  <CelText color={STYLES.COLORS.CELSIUS_BLUE} type={"H5"}>
```
Comment (fjovakaric) [inline app/components/molecules/CollateralCoinCard/CollateralCoinCard.js:106]: use new colors

File: app/components/molecules/CollateralLoanCard/CollateralLoanCard.js [hunk 8]
```diff
@@ -47,25 +47,31 @@ function getPropsFromTransaction(transaction) {
         color: getColor(COLOR_KEYS.POSITIVE_STATE),
       };
     case TRANSACTION_TYPES.COLLATERAL_UNLOCKED:
-      if (transaction.loan_data.unlock_reason === "rejected") {
+      if (transaction.loan_data.status === "rejected") {
```
Comment (fjovakaric) [inline app/components/molecules/CollateralLoanCard/CollateralLoanCard.js:50]: move to loan statuses constant

File: app/components/molecules/LoanCard/LoanCard.js [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+
+  static defaultProps = {};
+
+  constructor(props) {
+    super(props);
+    let data;
+
+    if (props.transaction) {
+      data = {
+        interestAmount: props.transaction.loan_data.loan_amount,
```
Comment (fjovakaric) [inline app/components/molecules/LoanCard/LoanCard.js:31]: don't use `props.` use `{ transaction } = props`

File: app/components/molecules/LoanCard/LoanCard.js [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+      data,
+    };
+  }
+
+  getPropsFromTransaction = transaction => {
+    switch (transaction.type) {
+      case TRANSACTION_TYPES.MARGIN_CALL:
+        return {
+          status: "Active Loan",
+          color: STYLES.COLORS.CELSIUS_BLUE,
```
Comment (fjovakaric) [inline app/components/molecules/LoanCard/LoanCard.js:55]: use new colors

File: app/components/molecules/LtvCard/LtvCard.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+  static defaultProps = {};
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      info: false,
+    };
+  }
+
+  set = () => {
```
Comment (fjovakaric) [inline app/components/molecules/LtvCard/LtvCard.js:27]: unused method

File: app/components/molecules/PaymentCard/PaymentCard.js [hunk 12]
```diff
@@ -1,151 +1,549 @@
...
-    name: PropTypes.string,
-    loanCollateral: PropTypes.number,
-    coinAmount: PropTypes.number,
+    coin: PropTypes.instanceOf(Object).isRequired,
+    type: PropTypes.oneOf("Interest", "PrincipalPayment", "MarginCall"),
+    isLoading: PropTypes.bool,
+    isMarginCall: PropTypes.bool,
+    marginCall: PropTypes.instanceOf(Object),
+    amountNeededUsd: PropTypes.string,
+    reason: PropTypes.oneOf(
```
Comment (fjovakaric) [inline app/components/molecules/PaymentCard/PaymentCard.js:46]: move to constants

File: app/components/molecules/PaymentCard/PaymentCard.js [hunk 13]
```diff
@@ -1,151 +1,549 @@
...
+
+  renderDepositMore = () => {
+    const { coin, actions, marginCall, reason } = this.props;
+    const { additionalUsdAmount, additionalCryptoAmount } = this.state;
+    return (
+      <View>
+        <CelButton
+          margin={"20 0 10 0"}
+          onPress={() =>
+            actions.navigateTo("Deposit", {
```
Comment (fjovakaric) [inline app/components/molecules/PaymentCard/PaymentCard.js:184]: SCREENS.DEPOSIT

File: app/components/molecules/PaymentCard/PaymentCard.js [hunk 14]
```diff
@@ -1,151 +1,549 @@
...
+              amountUsd: additionalUsdAmount,
+              additionalCryptoAmount,
+            })
+          }
+        >
+          {`Deposit more ${coin.short}`}
+        </CelButton>
+        {marginCall ? (
+          <CelButton
+            onPress={() => actions.navigateTo("Deposit", { coin: coin.short })}
```
Comment (fjovakaric) [inline app/components/molecules/PaymentCard/PaymentCard.js:196]: SCREENS.DEPOSIT

File: app/components/organisms/LoanOverviewCard/LoanOverviewCard.js [hunk 15]
```diff
@@ -106,121 +102,150 @@ class LoanOverviewCard extends Component {
...
       <View
         style={[
           style.container,
           getMargins(this.getMarginForIndex(index, length)),
         ]}
       >
+        {loan.can_pay_principal && (
+          <Card>
+            <CelText>You have completed all of your interest payments</CelText>
+            {/* <CelButton*/}
```
Comment (fjovakaric) [inline app/components/organisms/LoanOverviewCard/LoanOverviewCard.js:118]: remove comments

File: app/components/organisms/LoanOverviewCard/LoanOverviewCard.js [hunk 16]
```diff
@@ -236,61 +261,65 @@ class LoanOverviewCard extends Component {
...
-                    <View style={[style.interest, { marginBottom: 8 }]}>
-                      <CelText align={"center"} type={"H6"} weight={"300"}>
+                    <View
+                      style={[style.interest, style.additionalInterestStyle]}
+                    >
+                      <CelText align={"left"} type={"H6"} weight={"300"}>
                         Monthly interest
                       </CelText>
-                      <CelText align={"center"} type={"H3"} weight={"600"}>
+                      <CelText align={"left"} type={"H3"} weight={"600"}>
```
Comment (fjovakaric) [inline app/components/organisms/LoanOverviewCard/LoanOverviewCard.js:314]: is left alignment is default?

File: app/components/organisms/LoanOverviewCard/LoanOverviewCard.js [hunk 17]
```diff
@@ -304,19 +333,44 @@ class LoanOverviewCard extends Component {
...
+                  >
+                    {`You could have saved ${formatter.fiat(
+                      savedAmount,
+                      "USD"
+                    )} if you paid interest in CEL.`}
                   </CelText>
-                </Badge>
+                  <CelText
+                    onPress={() =>
+                      navigateTo("ChoosePaymentMethod", {
```
Comment (fjovakaric) [inline app/components/organisms/LoanOverviewCard/LoanOverviewCard.js:356]: use SCREENS

File: app/components/organisms/TierCard/TierCard.js [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+      ? Number(transaction.amount_usd)
+      : Number(loan.installments_to_be_paid.total);
+    const discountedInterest =
+      (1 - loyaltyInfo.tier.loanInterestBonus) * monthly;
+    const savedAmount = -1 * (monthly - discountedInterest);
+
+    let title;
+    let color;
+    if (loyaltyInfo.tier_level === 1) {
+      color = STYLES.COLORS.GRAY;
```
Comment (fjovakaric) [inline app/components/organisms/TierCard/TierCard.js:55]: use new colors in the component

File: app/components/screens/BorrowLoanTerm/BorrowLoanTerm.js [hunk 19]
```diff
@@ -69,6 +84,7 @@ class BorrowLoanTerm extends Component {
         <CelButton
           margin="50 0 30 0"
           onPress={() => {
+            // actions.navigateTo("BorrowLoanConfirm");
```
Comment (fjovakaric) [inline app/components/screens/BorrowLoanTerm/BorrowLoanTerm.js:87]: remove comments

File: app/components/screens/ChoosePaymentMethod/ChoosePaymentMethod.js [hunk 20]
```diff
@@ -117,20 +177,87 @@ class ChoosePaymentMethod extends Component {
...
+  };
+
+  automaticSwitch = () => {
+    const { isAutomaticInterestPaymentEnabled } = this.state;
+
+    const isIos = Platform.OS === "ios";
+    const falseColor = isIos ? "transparent" : STYLES.COLORS.DARK_GRAY3;
+    const theme = getTheme();
+    return (
+      <CelSwitch
```
Comment (fjovakaric) [inline app/components/screens/ChoosePaymentMethod/ChoosePaymentMethod.js:210]: use new colors

File: app/components/screens/ExtendLoanScreen/ExtendLoanScreen.js [hunk 21]
```diff
@@ -0,0 +1,125 @@
...
+
+    const sliderItems = monthValues.map(m => ({
+      value: m,
+      label: (
+        <>
+          <CelText
+            type="H6"
+            weight="bold"
+            color={
+              formData.extendPeriod === m ? STYLES.COLORS.CELSIUS_BLUE : null
```
Comment (fjovakaric) [inline app/components/screens/ExtendLoanScreen/ExtendLoanScreen.js:66]: use new colors

File: app/components/screens/InterestPaymentOverviewScreen/InterestPaymentOverviewScreen.js [hunk 22]
```diff
@@ -0,0 +1,119 @@
...
+                    </CelText>
+                  </View>
+                ))}
+              </View>
+
+              {!payment.hasEnough && (
+                <View style={{ marginHorizontal: 10 }}>
+                  <AdditionalAmountCard
+                    margin={"0 0 10 0"}
+                    color={COLOR_KEYS.NEGATIVE_STATE}
```
Comment (fjovakaric) [inline app/components/screens/InterestPaymentOverviewScreen/InterestPaymentOverviewScreen.js:89]: do we need getColor?

File: app/components/screens/InterestPaymentOverviewScreen/InterestPaymentOverviewScreen.js [hunk 23]
```diff
@@ -0,0 +1,119 @@
...
+                    additionalUsd={payment.additionalUsdAmount}
+                    additionalCryptoAmount={payment.additionalCryptoAmount}
+                  />
+                </View>
+              )}
+
+              <View style={style.buttonWrapper}>
+                <CelModalButton
+                  onPress={() => {
+                    actions.navigateTo("ChoosePaymentMethod", {
```
Comment (fjovakaric) [inline app/components/screens/InterestPaymentOverviewScreen/InterestPaymentOverviewScreen.js:101]: use SCREENS

File: app/components/screens/LoanPaymentCoin/LoanPaymentCoin.js [hunk 24]
```diff
@@ -73,20 +78,35 @@ class LoanPaymentCoin extends Component {
     }
 
     if (reason === LOAN_PAYMENT_REASONS.MANUAL_INTEREST) {
-      actions.navigateTo(SCREENS.VERIFY_PROFILE, {
-        onSuccess: () => actions.payMonthlyInterest(id, coinShort),
-      });
+      await actions.updateFormFields({ coin: coinShort });
+      actions.openModal(MODALS.CONFIRM_INTEREST_PAYMENT);
+      // actions.navigateTo(SCREENS.VERIFY_PROFILE, {
```
Comment (fjovakaric) [inline app/components/screens/LoanPaymentCoin/LoanPaymentCoin.js:83]: remove comments

File: app/components/screens/LoanPrepaymentPeriod/LoanPrepaymentPeriod.js [hunk 25]
```diff
@@ -52,12 +56,13 @@ class LoanPrepaymentPeriod extends Component {
...
         actions.navigateTo(SCREENS.WIRING_BANK_INFORMATION);
       } else {
-        actions.navigateTo(SCREENS.VERIFY_PROFILE, {
-          onSuccess: () => {
-            actions.prepayInterest(id);
-            actions.updateFormField("prepayLoanId", id);
-          },
-        });
+        actions.openModal(MODALS.CONFIRM_INTEREST_PREPAYMENT);
+        // actions.navigateTo(SCREENS.VERIFY_PROFILE, {
```
Comment (fjovakaric) [inline app/components/screens/LoanPrepaymentPeriod/LoanPrepaymentPeriod.js:60]: remove comments

File: app/components/screens/LoanSettings/LoanSettings.js [hunk 26]
```diff
@@ -51,16 +51,13 @@ class LoanSettings extends Component {
...
-            margin={"20 0 0 0"}
-            onPress={() =>
-              actions.navigateTo(SCREENS.PRINCIPAL_PAYMENT, { id })
-            }
-          >
-            Principal Payment
-          </IconButton>
-        )}
+
+        {/* <IconButton*/}
```
Comment (fjovakaric) [inline app/components/screens/LoanSettings/LoanSettings.js:55]: remove comments

File: app/components/screens/MarginCallOverviewScreen/MarginCallOverviewScreen.styles.js [hunk 27]
```diff
@@ -0,0 +1,25 @@
...
+
+const base = {
+  container: {
+    flex: 1,
+  },
+};
+
+const themed = {
+  light: {
+    card: { color: "#F3F3F3" },
```
Comment (fjovakaric) [inline app/components/screens/MarginCallOverviewScreen/MarginCallOverviewScreen.styles.js:13]: use new colors

File: app/components/screens/SingleMarginCallScreen/SingleMarginCallScreen.styles.js [hunk 28]
```diff
@@ -0,0 +1,25 @@
...
+
+const base = {
+  container: {
+    flex: 1,
+  },
+};
+
+const themed = {
+  light: {
+    card: { color: "#F3F3F3" },
```
Comment (fjovakaric) [inline app/components/screens/SingleMarginCallScreen/SingleMarginCallScreen.styles.js:13]: use new colors

File: app/components/screens/TransactionsIntersection/TransactionsIntersection.js [hunk 29]
```diff
@@ -44,11 +44,11 @@ class TransactionsIntersection extends Component {
   }
 
   static navigationOptions = ({ navigation }) => {
-    const hideBack = navigation.getParam("hideBack");
+    const hide = navigation.getParam("hideBack");
```
Comment (fjovakaric) [inline app/components/screens/TransactionsIntersection/TransactionsIntersection.js:47]: what was wrong with the previous code?

File: app/redux/loans/loansActions.js [hunk 30]
```diff
@@ -62,6 +64,14 @@ function applyForALoan() {
       const res = await loansService.apply(loanApplication, verification);
       dispatch({ type: ACTIONS.APPLY_FOR_LOAN_SUCCESS });
 
+      dispatch(
+        navigateTo("LoanRequestDetails", {
```
Comment (fjovakaric) [inline app/redux/loans/loansActions.js:68]: use SCREENS

File: app/redux/loans/loansActions.js [hunk 31]
```diff
@@ -390,25 +392,21 @@ function payPrincipal(id) {
...
-      const res = await loansService.lockMarginCallCollateral(
-        id,
-        coin,
-        verification
-      );
+      const res = await loansService.lockMarginCallCollateral(id, coin);
 
+      dispatch(closeModal());
       const transactionId = res.data.transaction_id;
+      // dispatch(
```
Comment (fjovakaric) [inline app/redux/loans/loansActions.js:404]: remove comments

File: app/redux/loans/loansActions.js [hunk 32]
```diff
@@ -483,6 +478,16 @@ function checkForLoanAlerts() {
         loanAlerts.push({ id: l.id, type: LOAN_ALERTS.MARGIN_CALL_ALERT });
       }
 
+      // if (
```
Comment (fjovakaric) [inline app/redux/loans/loansActions.js:481]: remove comments

File: app/redux/loans/loansActions.js [hunk 33]
```diff
@@ -563,3 +586,23 @@ function startedLoanApplication() {
...
   };
 }
+
+function extendLoan(id, numberOfMonths) {
+  return async dispatch => {
+    dispatch(startApiCall(API.EXTEND_LOAN));
+    try {
+      await loansService.extendLoan(id, numberOfMonths);
+      getAllLoans();
+      dispatch(navigateTo("BorrowLanding"));
```
Comment (fjovakaric) [inline app/redux/loans/loansActions.js:596]: use screens

File: app/redux/loans/loansActions.js [hunk 34]
```diff
@@ -563,3 +586,23 @@ function startedLoanApplication() {
...
+      dispatch(
```
Comment (fjovakaric) [inline app/redux/loans/loansActions.js:597]: add dispatch success action for api call

File: app/redux/user/userDataActions.js [hunk 35]
```diff
@@ -66,7 +66,8 @@ function linkBankAccount(bankAccountInfo) {
       const bankRes = await userDataService.linkBankAccount(bankAccountInfo);
       dispatch({ type: ACTIONS.LINK_BANK_ACCOUNT_SUCCESS });
       dispatch(updateFormField("bankInfo", bankRes.data));
-      dispatch(navigateTo(SCREENS.CONFIRM_YOUR_LOAN));
+      dispatch(navigateTo("BorrowLoanConfirm"));
+      // dispatch(navigateTo(SCREENS.CONFIRM_YOUR_LOAN));
```
Comment (fjovakaric) [inline app/redux/user/userDataActions.js:70]: use SCREENS

File: app/utils/loanPayment-util.js [hunk 36]
```diff
@@ -0,0 +1,185 @@
...
+import formatter from "./formatter";
+import { COLOR_KEYS } from "../constants/COLORS";
+
+const loanPaymentUtil = {
+  calculateAdditionalPayment,
+  isPrincipalWeekAway,
+  isSameInterestDay,
+};
+
+function calculateAdditionalPayment(
```
Comment (fjovakaric) [inline app/utils/loanPayment-util.js:14]: valjda radi, srecno sa testovima

**Review by fjovakaric — CHANGES_REQUESTED** [review]

File: app/components/modals/LoanAlertsModals/LoanAlertsPayoutPrincipalModal/LoanAlertsPayoutPrincipalModal.js (hunk 1 above)
Comment (ShamanShaulic) [reply app/components/modals/LoanAlertsModals/LoanAlertsPayoutPrincipalModal/LoanAlertsPayoutPrincipalModal.js:140]: This should be uncommented any time now

**Review by fjovakaric — APPROVED** [review]

**Pull request merged**